design/unnumbered-bgp.md

```diff
@@ -0,0 +1,498 @@
+              C3[eth0]
+            end
+        end
+
+        %% Connections between subgraphs
+        A3 -- point-to-point ------ B1[eth0]
+        A4 -- point-to-point ------ C1[eth0]
+    end
+
+    %% Define styles for subgraphs and nodes
+    classDef node fill:#f0f0f0,stroke:#000000,stroke-width:1px;
+    class Node0 node;
+    class Node1 node;
+```
+
+#### Story 2 - Unnumbered {i,E}BGP on secondary node interface
+
+```mermaid
+graph TD
+    subgraph Rack0
+        direction LR
+        style Rack0 fill:#ffffff,stroke:#000000,stroke-width:2px
+
+        subgraph ToR0
+            style ToR0 fill:#ffffff,stroke:#000000,stroke-width:2px
+            direction TB
+            subgraph Default
+              direction LR
+              A1[net0] -.- A{Routing}
+              A -.- A3[eth00]
+              A -.- A4[eth01]
+            end
+            subgraph RED VRF*
+              direction LR
+              AA1[net00] -.- AA{Routing}
+              AA -.- AA3[eth00.vlanRED]
+              AA -.- AA4[eth01.vlanRED]
+              BGP-T0(BGP configured on RED)
+            end
+
+        end
+
+        subgraph Node0
+            direction TB
+            B1 --- B2(cni)
+            B2 --- B3
+            BB1
+            subgraph Pod1[pod]
+              B3[eth0]
+            end
+            BGP-N0(BGP process)
+        end
+
+        subgraph Node1
+            direction TB
+            C1 --- C2(cni)
+            C2 --- C3
+            CC1
+            subgraph Pod2[pod]
+              C3[eth0]
+            end
+            BGP-N1(BGP process)
+        end
+
+        %% Connections between subgraphs
+        A3 ------- B1[eth0]
+        AA3 -- BGP Peering ----- BB1[net0]
+        A4 ------- C1[eth0]
+        AA4 -- BGP Peering ----- CC1[net0]
+    end
+
+    %% Define styles for subgraphs and nodes
+    classDef node fill:#f0f0f0,stroke:#000000,stroke-width:1px;
+    class Node0 node;
+    class Node1 node;
+```
+
+## Design details
+
+The `show running-config` in the FRR process running in the k8s node
+looks like
+
+
+```
+interface net0
+  ipv6 nd ra-interval 10
+  no ipv6 nd suppress-ra
+router bgp 65000
+  no bgp ebgp-requires-policy
+  no bgp network import-check
+
+  neighbor net0 interface remote-as 65004
+  neighbor net0 description TOR
+  neighbor net0 port 179
+  neighbor net0 bfd
+  neighbor net0 bfd profile fornet0
+  neighbor net0 timers 180 540
+  neighbor net0 timers connect 30
+  neighbor net0 graceful-restart
+  !  neighbor net0 password "xyz"
+
+  address-family ipv6 unicast
+    network fc00:f888:ccd:e793::1/128
+    neighbor net0 activate
+ exit-address-family
+  address-family ipv4 unicast
+    network 5.5.5.5/32
+    neighbor net0 activate
+ exit-address-family
+```
+
+* keep the format as in [neighborsession.tmpl](internal/bgp/frr/templates/neighborsession.tmpl)
+* instead of address (ip) to be interface name
+* enable RAs in that interface name
+* must exclude FRR directives that do not accept interface e.g.
```

Comment:
I found that this directive `disable-connected-check` can be combined with inteface. So it must be excluded otherwise throughs errors. I have not find other directive

Reply:
not sure I follow, are there options we can't use for unnumbered or it just needs a different configuration option?